File: src/main/java/com/task/twinotask/util/DateUtils.java
package com.task.twinotask.util;

import java.sql.Date;
import java.util.Calendar;

public class DateUtils {

	public static int yearsSince(Date pastDate) {
		Calendar present = Calendar.getInstance();
		Calendar past = Calendar.getInstance();
		past.setTime(pastDate);

		int years = 0;

		while (past.before(present)) {
			past.add(Calendar.YEAR, 1);
			if (past.before(present)) {
				years++;
			}
		}
		return years;
	}
}

File: src/main/kotlin/com/task/twinotask/util/DateUtils.kt
package com.task.twinotask.util

import java.sql.Date
import java.util.*

fun Date.yearsSince() : Int {
	val present = Calendar.getInstance()
	val past = Calendar.getInstance()
	past.time = this

	var years = 0

	while (past.before(present)) {
		past.add(Calendar.YEAR, 1)
		if (past.before(present)) ++years
	}
	return years
}
